code/controllers/subsystem/xenoarchaeology.dm

```diff
@@ -0,0 +1,216 @@
+SUBSYSTEM_DEF(xenoarchaeology)
+	name = "Xenoarchaeology"
+	flags = SS_NO_FIRE
+	init_order = INIT_ORDER_XENOARCHAEOLOGY
+
+	///Which console is the main character
+	var/obj/machinery/computer/xenoarchaeology_console/main_console
+
+	//All the traits - before we sort them, semi needed for generation
+	var/list/xenoa_all_traits = list()
+	///All traits indexed by name, used for labelling stuff
+	var/list/xenoa_all_traits_keyed = list()
+
+	///Names for science sellers & artifacts
+	var/list/xenoa_seller_names = list()
+	var/list/xenoa_seller_dialogue = list()
+	var/list/xenoa_artifact_names = list()
+
+	///Whitelist for traits by material type
+	var/list/material_traits = list()
```

Comment:
this isn't `traits`. this should be `material_info_datums` or `material_info_list`, smh

Reply:
breh